test/object_store/test_basic.py

```diff
@@ -60,7 +60,6 @@ def run_with_dir(run_cmd_gen, run_dir):
         # child process, not the ones used by the test.
         outputs = [(sys.stdout, 1), (sys.stderr, 2)]
         for output, output_fd in outputs:
-            output.flush()
```

Comment:
I explained in the other issue why I did this extra flush, though I admit it's not very important - I wanted the error message that might have been printed since the fork to appear on the original stdout and not in the redirected file.

NOTE: if you do leave this flush(), you need to flush the correct file! If sys.stdout doesn't point to fd 1, it's pointless to flush sys.stdout. So I agree, you can just remove this flush and it's fine.

Reply:
thank you for clarifying this. i amended the commit message accordingly.

Reply:
Actually it turned out that `run_scylla_cmd` commandline generator print()-s after fork() and this message should better be flushed(). So removing flushes is not correct :( Sorry for the noise

Reply:
ack. my fault. should have read `run_scylla_cmd()` before removing this flush call.

Reply:
but removing the flush still makes sense. as it well does not make sense as `run_scylla_cmd()` is printing to the buffer, and we can hardly do anything to grab that line of message out of the buffer. -- even we could using the fixture provided by pytest, it's overkill and does not offer much benefit.

Reply:
> and we can hardly do anything to grab that line of message out of the buffer. 

But why is that? The `run_scylla_cmd()` puts message into buffer. _Before_ we redirect stdios we can flush those and make the message into whatever stdio the _parent_ process thought this message had to be put into

Reply:
> > and we can hardly do anything to grab that line of message out of the buffer.
> 
> But why is that? The `run_scylla_cmd()` puts message into buffer.

to be more specific, the buffer here is the io buffer inherited from the parent process.

> _Before_ we redirect stdios we can flush those

yes, we can. but the thing is. if we are really interested in them after flushing them, we will need to write them to somewhere, like a file or stdout/stderr.

> and make the message into whatever stdio the _parent_ process thought this message had to be put into

yeah, that's the point. how can we get access to the io buffer the parent process redirects to? well, some of us would argue "we can", but i don't think it deserves the efforts to make it work. personally, i don't want to take too much time reading pytest's source code at this moment.

but after a second thought, i think the `flush` call still serve a purpose under a certain use case where we pass `-s` to pytest.

Reply:
i dropped the 2nd commit anyway. actually, it never belong to this PR in the first place..